index.js

```diff
@@ -129,7 +198,19 @@ function filterByGenre() {}
       }
     ];
  */
-function getAllMoviesReleasedAtOrBeforeYear() {}
+function getAllMoviesReleasedAtOrBeforeYear(movies, year) {
+  let releaseYearMovies = [];
+
+  for (let i = 0; i < movies.length; i ++){
+    let releasedArr = movies[i].released.split(' ')
+
+    if(year >= Number(releasedArr[releasedArr.length-1])){
+      releaseYearMovies.push(movies[i]);
+    }
+  }
```

Comment:
I would actually recommend on line 205 declaring a number variable such as `releasedYear` and setting it equal ` movies[i].released.split(' ')[movies[i].released.split(' ').length-1]`. It's clunky and long but at least we get all of the code out of the way on one line.  By splitting to `releasedArr` and then indexing we're creating an array var when all we want is the year value to compare to.